test/unit/cmap/message_stream.test.js

```diff
@@ -19,6 +19,34 @@ function bufferToStream(buffer) {
 }
 
 describe('Message Stream', function () {
+  context('when the stream uses the streaming protocol', function () {
+    const data = Buffer.from(
+      '370000000100000001000000010000000000000000000000000000000000000001000000130000001069736d6173746572000100000000' +
+        '370000000100000001000000010000000000000000000000000000000000000001000000130000001069736d6173746572000100000000' +
+        '370000000100000001000000010000000000000000000000000000000000000001000000130000001069736d6173746572000100000000' +
+        '370000000100000001000000010000000000000000000000000000000000000001000000130000001069736d6173746572000100000000',
+      'hex'
+    );
+
+    it('only reads the last message in the buffer', function (done) {
+      const inputStream = bufferToStream(data);
+      const messageStream = new MessageStream();
+      messageStream.isStreamingProtocol = true;
+
+      messageStream.once('message', msg => {
```

Comment:
instead of `done`, I think we can use async await here to make the test and the assertions more straightforward via the `once` from node events module, i.e. `await once(messageStream, 'message')`

Reply:
Updated to async/await